libs/wingc/src/parser.rs

```diff
@@ -731,28 +736,87 @@ impl<'s> Parser<'s> {
 			return module;
 		}
 
-		if module_name.name.starts_with("\"") && module_name.name.ends_with("\"") {
+		if module_symbol.name.starts_with("\"") && module_symbol.name.ends_with("\"") {
+			// we need to inspect the npm dependency to figure out if it's a JSII library or a Wing library
+			// first, find where the package.json is located
+			let module_name = module_symbol.name[1..module_symbol.name.len() - 1].to_string();
+			let source_dir = Path::new(&self.source_name).parent().unwrap();
+			let module_dir =
+				wingii::util::package_json::find_dependency_directory(&module_name, &source_dir).ok_or_else(|| {
+					self
+						.with_error::<Node>(
+							format!(
+								"Unable to load \"{}\": Module not found in \"{}\"",
+								module_name, self.source_name
+							),
+							&statement_node,
+						)
+						.err();
+				})?;
+
+			// If the package.json has a `wing` field, then we treat it as a Wing library
+			if is_wing_library(&Path::new(&module_dir)) {
+				return if let Some(alias) = alias {
+					let root_files = get_wing_library_root_files(&Path::new(&module_dir));
+
+					// concatenate the files' contents together
+					let mut source = String::new();
+					for file in &root_files {
+						source.push_str(&fs::read_to_string(file).unwrap_or(String::new()));
+					}
+
+					// generate a file name for their concatenation
+					let fake_path = PathBuf::from(module_dir).join("$root.w");
+
+					// track that the current file depends on the concatenated file, so it will get parsed downstream
+					self
+						.referenced_wing_files
+						.borrow_mut()
+						.push(WingSourceFile(fake_path.clone(), source));
```

Comment:
So, this file concatenation strategy does "work" (air quotes), but it's not a great solution since we're fudging up the line numbers from original source files, which means the LSP and error messages could get filled with incorrect information. 😕 

Need to find a better solution than this...